<!DOCTYPE html>
<html>

<head>
    <title>PitViper GUI</title>
    <meta content="text/html;charset=utf-8" http-equiv="Content-Type">
    <meta content="utf-8" http-equiv="encoding">
    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.0.0-beta2/dist/css/bootstrap.min.css" rel="stylesheet"
        integrity="sha384-BmbxuPwQa2lc/FVzBcNJ7UAyJxM6wuqIj61tLrc4wSX0szH/Ev+nYRRuWlolflfl" crossorigin="anonymous">
    <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.0.0-beta2/dist/js/bootstrap.bundle.min.js"
        integrity="sha384-b5kHyXgcpbZJO/tY9Ul7kGkf1S0CWuKcCD38l8YkeH8z8QjE0GmW1gYU5S9FOnJ0"
        crossorigin="anonymous"></script>
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
    <style ype=text/css scoped>
        h2 {
            margin-top: 1em;
            margin-bottom: 1em;
        }
    </style>
</head>

<body>
    <nav class="navbar navbar-expand-lg navbar-dark bg-dark px-5">
        <a class="navbar-brand" href="https://github.com/lobrylab/PitViper">PitViper GUI</a>
        <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarNavAltMarkup"
            aria-controls="navbarNavAltMarkup" aria-expanded="false" aria-label="Toggle navigation">
            <span class="navbar-toggler-icon"></span>
        </button>
        <div class="collapse navbar-collapse" id="navbarNavAltMarkup">
            <div class="navbar-nav">
                <a class="nav-item nav-link" href="/">Home</a>
                <a class="nav-item nav-link active" href="/documentation" target="_blank"><i class="fa fa-book"></i>
                    Documentation</a>
                <a class="nav-item nav-link" href="https://github.com/lobrylab/PitViper" target="_blank"><i
                        class="fa fa-github"></i>
                    Github</a>
            </div>
        </div>
    </nav>

    <div style="padding-left: 10%; padding-right: 10%;">
        <div class="paragraphs">
            <div class="row">
                <div class="span4 py-5 px-5">
                    <img style="float:left" src="/static/pitviper_remastered.png" width="507" height="175" />
                </div </div>
            </div>

            <p class="py-3  px-5">Welcome to the PitViper graphical user-interface documentation.</p>

            <p class="py-3  px-5">PitViper is intended to facilitate the analysis of functional screening data from
                various functionnal screening experiments (shRNA, CRISPR/Cas9 or CRISPR/dCas9).
            </p>

            <h1>How to use the PitViper GUI?</h1>

            <h2>General settings</h2>

            <h3>Name of the analysis</h3>
            <p class="py-3 px-5">The first step is to define the name of the analysis. This name is used to identify the
                results after the analysis is complete. Several directories are created during the analysis, the first
                one is located in the 'results/' directory and contains all the results from the analysis. The second
                one is located in the 'resources/' directory and contains all files necessary to run the analysis or
                reproduce the results. Finally, a configuration file is created and named after the analysis name,
                inside the 'config' directory. Therefore, we recommend using a informative name that characterize the
                analysis. </p>

            <h3>Screen type</h3>
            <p class="py-3 px-5">The screen type parameter specifies which options are available in the pipeline.
                Choosing the 'Not genes' screen type will disable and hide options incompatible with this setting. For
                example, when 'Not genes' is selected BAGEL2 is disabled from the tool panel but a file upload field is
                created in the Inputs panel to let users select BED annotation file in order to map genes to the
                elements of the screen.</p>

            <h3>Starting file format</h3>
            <p class="py-3 px-5">The starting file format parameter specifies the file format of the input file(s). If
                starting from FASTQ files, several fields will appear in the 'Inputs' and 'Tools' panels. Those fields
                are necessary to run the pipeline from this file format.</p>

            <h3>Jobs</h3>
            <p class="py-3 px-5">The jobs parameter specifies the number of jobs to run in parallel. This parameter is
                used to run the pipeline in parallel. The number of jobs is limited to the number of cores available on
                the machine running the pipeline. The default value is 1.</p>

            <h3>sgRNA filtering</h3>
            <p class="py-3 px-5">The sgRNA filtering parameter specifies the minimal median coverage of sgRNAs to keep
                in the analysis. The default value is 0. This parameter is used to filter sgRNAs with a low coverage in
                all conditions.</p>

            <h2>Inputs</h2>

            <h3>Design file</h3>
            <p class="py-3 px-5"> The design file is a tab-separated file containing the information about the
                experimental design. The first column contains the conditions names. The second column contains the
                replicates names and the third column contains the orderings of the conditions. The fourth column is
                mandatory only if the analysis begin with FASTQ files. It contains the path to the FASTQ file for each
                of the replicates. Order column should contain the order of the conditions in the analysis (e.g. 0, 1,
                2, etc.). All conditions will be used as a reference condition for essentiality analysis against all
                conditions with a higher order. For exemple, if the order column contains 0, 1, 2, all '0' conditions
                will be used as a reference condition for essentiality analysis against all '1' and '2' conditions.
                Then, '1' conditions will be used as a reference condition for essentiality analysis against all '2'
                conditions, but not against '0'.</p>

            <h3>Counts file</h3>
            <p class="py-3 px-5">The counts file is a tab-separated file containing the raw counts of sgRNAs in each
                condition. The first column contains the sgRNAs names. The second column contains the gene name. The
                third column contains the counts of the sgRNAs in the first condition. The fourth column contains the
                counts of the sgRNAs in the second condition, etc. The number of columns should be equal to the number
                of conditions in the design file plus two (sgRNA and gene columns). Columns names should match the
                replicates names in the design file.</p>

            <h3>Library file</h3>
            <p class="py-3 px-5">The library file is a tab-separated file containing the information about the
                sgRNAs. The first column contains the sgRNAs names. The second column contains the gene name. The third
                column contains the sgRNAs sequences. This file is mandatory if the analysis begin with FASTQ or BAM
                files.</p>

            <h3>Control guides file</h3>
            <p class="py-3 px-5">The control guides file contains the information about the control sgRNAs. More than
                100 guides is recommanded.</p>

            <h3>Elements coordinates</h3>
            <p class="py-3 px-5">The elements coordinates file contains the coordinates of the elements of the screen.
                This file is optional if the analysis type is 'Not genes' and useless if the analysis type is 'Genes'.
                This annotation file is used to map genes to the elements of the screen. For example, if the screen is a
                CRISPRi screen targeting enhancers, the elements coordinates file will contain the coordinates of the
                enhancers in the genome of reference. Then, the 'ROSE gene mapper' script is used to annotate each
                enhancers with proximal genes. Several genome versions are available: HG38, HG19 and HG18 for human,
                MM10, MM9 and MM8 for mouse.
            </p>

</html>
